File: LC1365.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        res = sorted(nums)
        d = {}
        
        for i in range(len(nums)):
            if res[i] not in d:
                d[res[i]] = i
        
        for i in range(len(nums)):
            res[i] = d[nums[i]]


        return res

File: LC1365.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        int n = nums.size(), count = 0;
        vector<int> res(nums);
        map <int, int> m;

        sort(res.begin(), res.end());

        for(int i = 0; i < n; i++){
            if(m.find(res[i]) ==  m.end())
                m[res[i]] = i;
        }

        for(int i = 0; i < n; i++)
            res[i] = m[nums[i]];

        return res;
    }
};
